### PR TITLE
add google_storage_notification resource

### LINKED ISSUE #169: Configure filestore (2), gke (2), iam (4), identityplatform (2), kms (1), logging (1), monitoring (2), scc (1), storage (6), vertexai (7)
Configure external names for the following resources:

* `filestore`
  * [x] [google_filestore_backup](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/filestore_backup)
   * [x] [google_filestore_snapshot](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/filestore_snapshot)

* `gke`
  * [x] [google_gke_backup_backup_plan](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_backup_backup_plan)
   * [x] [google_gke_hub_membership_iam_member](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_membership_iam)

* `iam`
  * [x] [google_iam_workforce_pool](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workforce_pool)
   * [x] [google_iam_workforce_pool_provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workforce_pool_provider)
    * [x] [google_iam_workload_identity_pool](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workload_identity_pool)
   * [x] [google_iam_workload_identity_pool_provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/iam_workload_identity_pool_provider)

* `identityplatform`
  * [x] [google_identity_platform_config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/identity_platform_config)
   * [x] [google_identity_platform_project_default_config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/identity_platform_project_default_config)

* `kms`
  * [x] [google_kms_crypto_key_version](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/kms_crypto_key_version)

* `logging`
  * [x] [google_logging_billing_account_bucket_config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/logging_billing_account_bucket_config)

* `monitoring`
  * [x] [google_monitoring_monitored_project](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/monitoring_monitored_project)
   * [x] [google_monitoring_service](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/monitoring_service)

* `scc`
  * [x] [google_scc_source_iam_member](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/scc_source_iam)

* `storage`
  * [x] [google_storage_hmac_key](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_hmac_key)
   * [x] [google_storage_notification](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_notification)
  * [x] [google_storage_object_access_control](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_object_access_control)
  * [x] [google_storage_object_acl](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_object_acl)
  * [x] [google_storage_transfer_agent_pool](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_transfer_agent_pool)
  * [x] [google_storage_transfer_job](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_transfer_job)

* `vertexai`
  * [x] [google_vertex_ai_dataset](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_dataset)
   * [x] [google_vertex_ai_endpoint](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_endpoint)
  * [x] [google_vertex_ai_featurestore](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_featurestore)
  * [x] [google_vertex_ai_featurestore_entitytype](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_featurestore_entitytype)
  * [x] [google_vertex_ai_featurestore_entitytype_feature](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_featurestore_entitytype_feature)
  * [x] [google_vertex_ai_index](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_index)
  * [x] [google_vertex_ai_tensorboard](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_tensorboard)

### PR DESCRIPTION
**Pull request by bradkwadsworth-mw**
Signed-off-by: Brad Wadsworth <brad.wadsworth@mavenwave.com>

### Description of your changes
Added the google_storage_notification resource.

Fixes #169 

I have:

- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Testing from a local cluster with the example at `examples/storage/notification.yaml`

Screenshot 2023-03-30 at 15 56 16

### CODE REVIEW

**Comment by bradkwadsworth-mw** [comment]
/test-examples="provider-gcp/examples/storage/notification.yaml"

**Comment by bradkwadsworth-mw** [comment]
/test-examples="examples/storage/notification.yaml"

**Comment by bradkwadsworth-mw** [comment]
/test-examples="examples/storage/notification.yaml"

**Comment by bradkwadsworth-mw** [comment]
/test-examples="examples/storage/notification.yaml"

**Comment by bradkwadsworth-mw** [comment]
/test-examples="examples/storage/notification.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="examples/storage/notification.yaml"

**Comment by bradkwadsworth-mw** [comment]
I think for this test to work, a `cloudplatform_project.projectNumber` entry needs to be added with the gcp project number to the datasource.yaml. This is required because the Pub/Sub Topic needs to allow `service-${data.cloudplatform_project.projectNumber}@gs-project-accounts.iam.gserviceaccount.com` `roles/pubsub.publisher` access.

**Comment by turkenf** [comment]
Require status checks changed could you please rebase this PR?

**Comment by turkenf** [comment]
/test-examples="examples/storage/notification.yaml"

**Comment by turkenf** [comment]
Hi @bradkwadsworth-mw, many thanks for this PR, could you please add a screenshot to the description that you tested it?
See example: https://github.com/upbound/provider-aws/pull/500#issue-1578645471

**Review by turkenf — COMMENTED** [review]
Thanks, @bradkwadsworth-mw for configuring and testing these resources, I left some comments. 
Also, ignore the screenshot I asked from you at the top, I tested it and added the screenshot.

File: config/storage/config.go [hunk 1]
```diff
@@ -39,4 +39,13 @@ func Configure(p *config.Provider) {
...
 	})
+
+	p.AddResourceConfigurator("google_storage_notification", func(r *config.Resource) {
+		r.References["bucket"] = config.Reference{
+			Type: "Bucket",
+		}
+		r.References["topic"] = config.Reference{
+			Type: "github.com/upbound/provider-gcp/apis/pubsub/v1beta1.Topic",
+		}
+	})
```
[the remark this answers is not in the document]
Comment (bradkwadsworth-mw) [reply config/storage/config.go]: I removed the extra config, didn't realize it was automatic.

File: examples/storage/notification.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+apiVersion: storage.gcp.upbound.io/v1beta1
+kind: Notification
+metadata:
+  annotations:
+    meta.upbound.io/example-id: storage/v1beta1/notification
```
[the remark this answers is not in the document]
Comment (bradkwadsworth-mw) [reply examples/storage/notification.yaml:5]: All set

File: examples/storage/notification.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+kind: TopicIAMMember
+metadata:
+  annotations:
+    meta.upbound.io/example-id: storage/v1beta1/notification
+  labels:
+    testing.upbound.io/example-name: topic
+  name: topic-iam-member
+spec:
+  forProvider:
+    member: "serviceAccount:service-${data.cloudplatform_project.projectNumber}@gs-project-accounts.iam.gserviceaccount.com"
```
[the remark this answers is not in the document]
Comment (bradkwadsworth-mw) [reply examples/storage/notification.yaml]: All set

**Review by turkenf — APPROVED** [review]

**Pull request merged**